plugins/woocommerce/src/Admin/RemoteInboxNotifications/RemoteInboxNotificationsEngine.php

```diff
@@ -63,6 +70,11 @@ function() {
 		);
 
 		add_filter( 'woocommerce_get_note_from_db', array( __CLASS__, 'get_note_from_db' ), 10, 1 );
+		add_filter( 'woocommerce_debug_tools', array( __CLASS__, 'add_debug_tools' ) );
+		add_action(
```

Comment:
We are trying to avoid adding public methods to existing classes whenever possible to reduce the backwards-compatibility surface of the code, especially when the method is public for the sole purpose of being the target of a hook like in this case. So could you please:

1. Add `use AccessiblePrivateMethods;`  inside the class (and also `use Automattic\WooCommerce\Internal\Traits\AccessiblePrivateMethods;` outside of the class if your editor doesn't add it automatically).
2. Change `add_filter` and `add_action` here to `self::add_filter` and `self::add_action`.
3. Make the `add_debug_tools` and `ajax_action_inbox_notification_search` methods private.

Thank you!

Reply:
@Konamiman Thank you! TIL.

Updated in [Use private function for the action and filter](https://github.com/woocommerce/woocommerce/pull/48961/commits/a9bba12d3d83252fb121bb7910eba7e302f16b2e)